File: demo/node/app/lib/email.js
function sendEmail(to, name, subject, text) {
  console.log(`[EMAIL] To: ${to} | Subject: ${subject} | Body:\n${text}`);
}

export function sendVerificationEmail(email, name, token) {
  const appName = process.env.APP_NAME || "Demo";
  const baseUrl = (process.env.BASE_URL || "http://localhost:3000").replace(/\/$/, "");
  const url = `${baseUrl}/api/auth/verify-email?token=${token}`;
  sendEmail(email, name, `Verify your ${appName} email`, `Use the link below to verify your ${appName} email address.\n\n${url}\n\nIf you did not create a ${appName} account, you can ignore this email.`);
}

export function sendPasswordResetEmail(email, name, token) {
  const appName = process.env.APP_NAME || "Demo";
  const baseUrl = (process.env.BASE_URL || "http://localhost:3000").replace(/\/$/, "");
  const url = `${baseUrl}/reset-password?token=${token}`;
  sendEmail(email, name, `Reset your ${appName} password`, `Use the link below to reset your ${appName} password.\n\n${url}\n\nIf you did not request this, you can ignore this email.`);
}

File: demo/go/internal/auth/email.go
package auth

import (
	"fmt"
	"log"
	"os"
	"strings"
)

var (
	baseURL string
)

func InitEmail(url string) {
	baseURL = url
}

func sendEmail(to, name, subject, text string) {
	log.Printf("[EMAIL] To: %s | Subject: %s | Body:\n%s", to, subject, text)
}

func sendVerificationEmail(email, name, token string) {
	appName := "Demo"
	if tamaya := os.Getenv("APP_NAME"); tamaya != "" {
		appName = tamaya
	}

	url := fmt.Sprintf("%s/api/auth/verify-email?token=%s", strings.TrimRight(baseURL, "/"), token)
	text := strings.Join([]string{
		"Use the link below to verify your " + appName + " email address.",
		"",
		url,
		"",
		"If you did not create a " + appName + " account, you can ignore this email.",
	}, "\n")

	sendEmail(email, name, "Verify your "+appName+" email", text)
}

func sendPasswordResetEmail(email, name, token string) {
	appName := "Demo"
	if tamaya := os.Getenv("APP_NAME"); tamaya != "" {
		appName = tamaya
	}

	url := fmt.Sprintf("%s/reset-password?token=%s", strings.TrimRight(baseURL, "/"), token)
	text := strings.Join([]string{
		"Use the link below to reset your " + appName + " password.",
		"",
		url,
		"",
		"If you did not request this, you can ignore this email.",
	}, "\n")

	sendEmail(email, name, "Reset your "+appName+" password", text)
}
